### DISCUSSION TITLE
What is system design? | System Design

### DISCUSSION THREAD

**Discussion by giscus** [karanpratapsingh/portfolio#35] [Comments]
# What is system design? | System Design

What is system design and why is it useful?

https://www.karanpratapsingh.com/courses/system-design/what-is-system-design

**Comment by Aztinct** [comment]
System design is about designing the system. Building the lego pieces that forms the infrastructure. Akin to building one's personal computer. In this case, the enterprise level computer parts, and the purpose of this high end enterprise PC. This will determine the parts necessary to suit the purpose. 

Like an sports car, it's high maintenance but it enables the driver to move really fast. Or, like a lorry, speed isn't important, but the ability to carry heavy loads is. One's goal is to transport a person, another is to transport goods. Different purposes. 

Considerations go beyond hardware and down to the software level and even how data is managed.
